src/templates/partials/lighttpd.hbs

```diff
@@ -11,36 +15,59 @@ $SERVER["socket"] == ":80" {
 {{/if}}
 }
 
+#server.modules += ("mod_setenv")
+$HTTP["scheme"] == "https" {
+    # HTTP Strict Transport Security ({{output.hstsMaxAge}} seconds)
+    setenv.add-response-header = (
+        "Strict-Transport-Security" => "max-age={{output.hstsMaxAge}}"
+    )
+}
 {{/if}}
+
+#server.modules += ("mod_openssl")
 $SERVER["socket"] == ":443" {
-    ssl.engine   = "enable"
+    ssl.engine  = "enable"
 
+  {{#if (minver "1.4.53" form.serverVersion)}}
+    ssl.privkey = "/path/to/private_key"
+    ssl.pemfile = "/path/to/signed_cert"
+    ssl.ca-file = "/path/to/intermediate_certificate"
+    # alternative: pemfile is cert+privkey, ca-file is the intermediate chain in one file
+    #ssl.pemfile = "/path/to/signed_cert_plus_private_key"
+    #ssl.ca-file = "/path/to/intermediate_certificate"
+  {{else}}
     # pemfile is cert+privkey, ca-file is the intermediate chain in one file
-    ssl.pemfile               = "/path/to/signed_cert_plus_private_key"
-    ssl.ca-file               = "/path/to/intermediate_certificate"
+    ssl.pemfile = "/path/to/signed_cert_plus_private_key"
+    ssl.ca-file = "/path/to/intermediate_certificate"
+  {{/if}}
 {{#if output.usesDhe}}
     {{#if (minver "1.4.29" form.serverVersion)}}
 
     # {{output.dhCommand}} > /path/to/dhparam
-    ssl.dh-file               = "/path/to/dhparam"
+    ssl.dh-file = "/path/to/dhparam"
     {{/if}}
 {{/if}}
 
     # {{form.config}} configuration
-{{#if (minver "1.4.48" form.serverVersion)}}
-    ssl.openssl.ssl-conf-cmd = ("Protocol" => "ALL, -SSLv2, -SSLv3{{#unless (includes "TLSv1" output.protocols)}}, -TLSv1{{/unless}}{{#unless (includes "TLSv1.1" output.protocols)}}, -TLSv1.1{{/unless}}{{#unless (includes "TLSv1.2" output.protocols)}}, -TLSv1.2{{/unless}}")
-{{else}}
-    # Please upgrade to 1.4.48 or else you cannot fully disable deprecated protocols
+ {{#if (minver "1.1.0" form.opensslVersion)}}
+  {{#if (minver "1.4.55" form.serverVersion)}}
```

Comment:
Why does this syntax not work with older versions?

Reply:
openssl 1.0.2 added API `SSL_CONF_cmd` including "Protocol"
lighttpd 1.4.48 added `ssl.openssl.ssl-conf-cmd`
openssl 1.1.0 added API `SSL_CONF_cmd` "MinProtocol"

I have adjusted the conditions to better reflect those interactions.